apollo-normalized-cache-api/src/commonMain/kotlin/com/apollographql/apollo3/cache/normalized/api/CacheResolver.kt

```diff
@@ -92,17 +95,53 @@ object DefaultCacheResolver: CacheResolver {
   ): Any? {
     val name = field.nameWithArguments(variables)
     if (!parent.containsKey(name)) {
+      @OptIn(ApolloInternal::class)
       throw CacheMissException(parentId, name)
     }
 
     return parent[name]
   }
 }
 
+
+/**
+ * A cache resolver that uses the parent to resolve fields and use a constant max A
+ */
+@ApolloExperimental
+class MaxAgeCacheResolver(private val maxAge: Int) : CacheResolver {
```

Comment:
I'm usually all for that but in that case I was thinking the similarity with the HTTP headers and existing [`@cacheControl` spec](https://www.apollographql.com/docs/apollo-server/performance/caching/) could maybe save the verbosity and make things symmetrical. Not sure...

Reply:
I went with `Long` for dates and `Int` for ages. Because ages are typically much smaller than dates and this is a hint that huge values aren't supported there.

Reply:
Got it 👍